src/Core/Checkout/Cart/CartPersister.php

```diff
@@ -101,6 +102,15 @@ public function delete(string $token, SalesChannelContext $context): void
         $this->connection->delete('`cart`', ['token' => $token]);
     }
 
+    private function isEmptyCart(Cart $cart): bool
+    {
+        return $cart->getLineItems()->count() <= 0
+            && empty($cart->getExtensions())
+            && $cart->getCustomerComment() === null
+            && $cart->getCampaignCode() === null
+            && $cart->getAffiliateCode() === null;
```

Comment:
transactions and deliveries should not be necessary because those are automatically created from the sales channel context during cart loading.

though I am not 100% sure regarding deliveries but from what I see the original are only kept if the skip recalculate permission is used which is not the case during a normal cart reloading process.